test/lib/ansible_test/_internal/docker_util.py

```diff
@@ -195,10 +195,20 @@ def docker_images(args, image):
     except SubprocessError as ex:
         if 'no such image' in ex.stderr:
             stdout = ''  # podman does not handle this gracefully, exits 125
+        elif 'function "json" not defined' in ex.stderr:
+            # podman > 2 && < 2.2.0 breaks with --format {{json .}}, and requires --format json
+            # So we try this as a fallback. If it fails again, we just raise the exception and bail.
+            stdout, _dummy = docker_command(args, ['images', image, '--format', 'json'], capture=True, always=True)
         else:
             raise ex
-    results = [json.loads(line) for line in stdout.splitlines()]
-    return results
+
+    lines = stdout.splitlines()
+    if lines and lines[0][0] == '[':
+        # modern podman outputs a pretty-printed json list. Just load the whole thing.
+        return json.loads(stdout)
+
+    # docker outputs one json object per line
+    return [json.loads(line) for line in lines]
```

Comment:
How about not attempting to scan the whole stdout for line feeds prematurely?
```suggestion
    is_jsonl = not stdout.startswith('[')
    if not is_jsonl:
        # modern podman outputs a pretty-printed json list. Just load the whole thing.
        return json.loads(stdout)

    # docker outputs one json object per line (a.k.a. JSONL)
    return [json.loads(json_str) for json_str in stdout.splitlines()]
```